### PR TITLE
OneHotEncoding sample

### PR DESCRIPTION
**Pull request by ganik**
Sample code for OneHotEncoding

### CODE REVIEW

File: docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Data;
+using static Microsoft.ML.Transforms.OneHotEncodingTransformer;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public static class OneHotEncodingTransform
+    {
+        public static void Example()
```
Comment (wschin) [inline docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs]: Does it come from an existing test?

File: docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            //
+            // Age  Case  Education  Induced     Parity  PooledStratum  RowNum  ...
+            // 26   1       0-5yrs      1         6         3             1  ...
+            // 42   1       0-5yrs      1         1         1             2  ...
+            // 39   1       0-5yrs      2         6         4             3  ...
+            // 34   1       0-5yrs      2         4         2             4  ...
+            // 35   1       6-11yrs     1         3         32            5  ...
+
+            // A pipeline for one hot encoding the Education column.
+            var pipeline = ml.Transforms.Categorical.OneHotEncoding("EducationOneHotEncoded", "Education", OutputKind.Bag);
```
Comment (wschin) [inline docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs]:
```suggestion
            var pipeline = ml.Transforms.Categorical.OneHotEncoding("EducationOneHotEncoded", "Education", OutputKind.Bag);
            
```

File: docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs (hunk 1 above)
Comment (sfilipi) [inline docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs]: >Example [](start = 27, length = 7)

please link it to the extension it documents through the <example> node. See the other extension methods xml doc.

File: docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    public static class OneHotEncodingTransform
+    {
+        public static void Example()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var ml = new MLContext();
+
+            // Get a small dataset as an IEnumerable and convert it to an IDataView.
+            IEnumerable<SamplesUtils.DatasetUtils.SampleInfertData> data = SamplesUtils.DatasetUtils.GetInfertData();
```
Comment (sfilipi) [inline docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs]: >IEnumerable<SamplesUtils.DatasetUtils.SampleInfertData> [](start = 12, length = 55)

var

Comment (sfilipi) [inline docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs]: >GetInfertData [](start = 101, length = 13)

think we want to deprecate the infert datasets from the samples; as it is a sensitive one. Is it possible to use one of the other dataset snippets?

File: docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                {
+                    for (var i = 0; i < row.Length; i++)
+                        Console.Write($"{row[i]} ");
+                    Console.WriteLine();
+                }
+            };
+
+            printHelper("Education", encodedColumn);
+
+            // data column obtained post-transformation.
```
Comment (sfilipi) [inline docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs]: `EducationOneHotEncoded` data column

**Comment by shmoradims** [comment]
> using System;

Please move this file to a path similar to it's access path from MLContext: Dynamic/Transforms/Categorical/OneHotEncoding.cs

---
Refers to: docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs:1 in daeba69. [](commit_id = daeba69ff872dc0e9f6e5f40cf98c305c74481bc, deletion_comment = False)

File: docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Data;
+using static Microsoft.ML.Transforms.OneHotEncodingTransformer;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public static class OneHotEncodingTransform
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs]: >OneHotEncodingTransform [](start = 24, length = 23)

Please rename this to OneHotEncoding (identical to the API extension method that it correspond to)

File: docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs (hunk 3 above)
Comment (shmoradims) [reply docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs]: you can use Adult dataset that exists in DatasetUtils

---
In reply to: [261049116](https://github.com/dotnet/machinelearning/pull/2779#discussion_r261049116) [](ancestors = 261049116)

File: docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            // A small printing utility.
+            Action<string, IEnumerable<float[]>> printHelper = (colName, column) =>
+            {
+                foreach (var row in column)
+                {
+                    for (var i = 0; i < row.Length; i++)
+                        Console.Write($"{row[i]} ");
+                    Console.WriteLine();
+                }
+            };
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs]: Consider moving this to SampleUtils.ConsoleUtils

File: docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            // data column obtained post-transformation.
+            // 1 0 0 0 ...
+            // 1 0 0 0 ... 
+            // 1 0 0 0 ...
+            // 1 0 0 0 ... 
+            // 0 1 0 0 ...
+            // ....
+        }
+    }
+}
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/OneHotEncodingTransform.cs]: please also add a sample for the overload version with ColumnOptions and call it OneHotEncodingWithOptions.cs

**Comment by sfilipi** [comment]
Work is moved to here: https://github.com/dotnet/machinelearning/pull/2960

**Pull request closed without merging**